CRAFTERCMS-2643

dejan-brkic opened the pull request:
fixed content type definition and upgrade

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2643

jdrossl committed:
Add new operations for upgrade manager

jdrossl committed:
Update version for db and blueprints

jdrossl committed:
Update documentation for upgrade operations

jdrossl committed:
Fix Codacy issue

Commit:
Content monitoring was using solr search

Commit:
fixed content type definition and upgrade

Commit:
fixed content type definition and upgrade

Commit:
Merge pull request #1466 from dejan-brkic/bugfix/2836

CRAFTERCMS-2836

jdrossl commented on src/main/resources/crafter/studio/upgrade/pipelines.yaml:
```diff
@@ -275,3 +275,12 @@ configurations:
             path: *site-config-tools
             template: crafter/studio/upgrade/update-version.xslt
+      - currentVersion: 2
+        nextVersion: 3
```
Please remember to update the version in the file for all blueprints

jdrossl reviewed (commented):
I think we also need to add upgrade operations to update the `sample-form-control-rte-setup.xml` & `rte-setup.xml` files according to the original PR https://github.com/craftercms/studio/pull/1365/files

jdrossl committed:
Add internal name for all search result items

jdrossl committed:
Fixes for search:
- sortOrder now accepts lowercase
- sortBy now tries to map to fields in the configured facets
- improve exception for errors to be handled as expected by spring

jdrossl committed:
Fix item names in content monitoring

jdrossl committed:
Update to allow a full Lucene query to be provided

Commit:
Fixed issue by escaping string in groovy pages

Commit:
Fixed issue by escaping string in groovy pages

Commit:
Merge pull request #1471 from dejan-brkic/bugfix/2817

CRAFTERCMS-2817

Commit:
Merge pull request #1470 from jdrossl/bugfix/2836

Fix for content monitoring search

Commit:
Merge pull request #1469 from jdrossl/bugfix/2849

Fixes for search:

Commit:
Merge pull request #1468 from jdrossl/feature/2850

Add internal name for all search result items

Commit:
Added requested changes

Commit:
Added requested changes

jdrossl commented on src/main/resources/crafter/studio/upgrade/pipelines.yaml:
```diff
@@ -209,6 +209,18 @@ pipelines:
         - type: versionFileUpgrader
 
+    - currentVersion: 3.1.0.3
+      nextVersion: 3.1.0.4
```
Please update `studio_version.xml` in the blueprints to match this version

jdrossl committed:
Fix typo in content monitoring configuration for all blueprints

Commit:
Updated docs

jdrossl committed:
Update logging to use Log4j2:
- Cleanup test resources & add system properties

Commit:
Merge pull request #1473 from jdrossl/bugfix/2836

Fix typo in content monitoring configuration for all blueprints

Commit:
Merge pull request #1474 from dejan-brkic/feature/2833

CRAFTERCMS-2833

jdrossl committed:
Refactor to remove log4j dependency from core library

Commit:
Fixed log issues when creating site in clustered environment

sumerjabri committed:
Merge pull request #1477 from dejan-brkic/bugfix/2818

CRAFTERCMS-2818

jdrossl committed:
Fix content-type metadata when uploading files to S3

jdrossl committed:
Fix search doc that was overwritten

sumerjabri committed:
Merge pull request #1480 from jdrossl/doc

Fix search doc that was overwritten

Commit:
Added missing preview sync when replicating site on cluster member

sumerjabri committed:
Syntax fixes

Fixed syntax/indentation issues. Now works with OAS 3/SwaggerHub

sumerjabri committed:
Merge pull request #1481 from dejan-brkic/bugfix/2861

CRAFTERCMS-2861

sumerjabri committed:
Merge pull request #1458 from jdrossl/feature/2646

New Upgrade Operations

Commit:
added upload asset to CMIS

Commit:
Only inactive members can be removed from cluster

jdrossl committed:
Fixes for search:
- Fix query for keywords
- Add support for range facets
- Update facets names

sumerjabri committed:
Merge pull request #1479 from jdrossl/bugfix/2859

Fix content-type metadata when uploading files to S3

sumerjabri committed:
Merge pull request #1482 from dejan-brkic/feature/2863

CRAFTERCMS-2863

sumerjabri committed:
Merge pull request #1483 from dejan-brkic/feature/2864

CRAFTERCMS-2864

sumerjabri committed:
Merge pull request #1485 from jdrossl/search-fixes

Fixes for search:

sumerjabri committed:
Update studio-api-2.yaml

sumerjabri committed:
Merge pull request #1476 from jdrossl/feature/2855

Update logging to use Log4j2:

jdrossl committed:
Refactor for search range facets configuration & api

sumerjabri commented:
@dejan-brkic bump

Commit:
Audit log API v2

Commit:
update version number

Commit:
requested changes

sumerjabri committed:
Merge pull request #1486 from dejan-brkic/feature/2870

CRAFTERCMS-2870

jdrossl committed:
Fixes for log4j2 support

jdrossl committed:
Fix log4j filter order

jdrossl committed:
Updates for search service:
- Use ui-friendly names for facets
- Add mime type for all result items

sumerjabri committed:
Merge pull request #1488 from jdrossl/feature/2874

Updates for search service:

sumerjabri committed:
Merge pull request #1487 from jdrossl/feature/2855

Fixes for log4j2 support

Commit:
Added permission to view audit log

vestrada committed:
https://github.com/craftercms/craftercms/issues/2869 - [studio-ui] Create a global audit viewing tool #2869

sumerjabri committed:
Merge pull request #1490 from dejan-brkic/feature/2870

CRAFTERCMS-2870

jdrossl committed:
Updates for search service:
- Implement override for search facets in site config
- Implement custom ranges for facets
- Delete an unused class

jdrossl committed:
Move logging configuration to extension folder

sumerjabri committed:
Merge pull request #1492 from jdrossl/feature/2855

Move logging configuration to extension folder

sumerjabri committed:
Update studio-api-2.yaml

vestrada committed:
https://github.com/craftercms/craftercms/issues/2869 - [studio-ui] Create a global audit viewing tool #2869

jdrossl committed:
Merge branch 'develop' of https://github.com/craftercms/studio into feature/2872

* 'develop' of https://github.com/craftercms/studio:
  Move logging configuration to extension folder
  Added permission to view audit log

# Conflicts:
#	src/main/api/2/studio-api-2.yaml

sumerjabri committed:
Merge pull request #1491 from jdrossl/feature/2872

Updates for search service

jdrossl committed:
Fix api doc

sumerjabri committed:
Merge pull request #1493 from jdrossl/feature/2872

Fix api doc

Commit:
Fixed clone site algorithm

Commit:
Fixed clone site algorithm

vestrada committed:
https://github.com/craftercms/craftercms/issues/2869 - [studio-ui] Create a global audit viewing tool #2869

Commit:
Merge branch 'bugfix/2881' of github.com:dejan-brkic/studio into bugfix/2881

sumerjabri committed:
Merge pull request #1489 from vestrada/feature/2869

#2869

jdrossl committed:
Updates for search service:
- Simplified global & site facets merging
- Fix sorting for range facets
- Fix range filters

Commit:
Fix sync bugs

sumerjabri committed:
Merge pull request #1494 from jdrossl/feature/2872

Updates for search service:

sumerjabri committed:
Merge pull request #1495 from dejan-brkic/bugfix/2881

CRAFTERCMS-2881

Commit:
Upgrade global-permission-mappings-config.xml

sumerjabri committed:
Merge pull request #1496 from dejan-brkic/bugfix/2893

CRAFTERCMS-2893

Commit:
Added missing listener to web.xml

sumerjabri committed:
Merge pull request #1497 from dejan-brkic/bugfix/2905

CRAFTERCMS-2905

Commit:
fixed parameter issues

sumerjabri committed:
Merge pull request #1498 from dejan-brkic/feature/2869

CRAFTERCMS-2869

Commit:
Fixed errors when creating a site named the same as a site that has been deleted

jdrossl committed:
Update default ranges for facets

sumerjabri committed:
Merge pull request #1499 from dejan-brkic/bugfix/2908

CRAFTERCMS-2908

sumerjabri committed:
Merge pull request #1500 from jdrossl/search-fixes

Update default ranges for facets

Commit:
Fixed wrong response code for invalid site

sumerjabri committed:
Merge pull request #1501 from dejan-brkic/bugfix/2916

CRAFTERCMS-2916

jdrossl committed:
Update monitoring API

jdrossl committed:
Allow regex for public urls in security configuration

jdrossl committed:
Update according to feedback

sumerjabri committed:
Merge pull request #1502 from jdrossl/feature/2875

Update monitoring API

Commit:
Use blueprint ID instead of folder name in create site API

Commit:
remove folder name field

Commit:
added screenshots

jvega190 committed:
[studio-ui] Upgrade to TinyMCE 5 #2839

sumerjabri committed:
Update craftercms-plugin.yaml

sumerjabri committed:
Update craftercms-plugin.yaml

sumerjabri committed:
Update craftercms-plugin.yaml

sumerjabri committed:
Update craftercms-plugin.yaml

sumerjabri committed:
Update craftercms-plugin.yaml

sumerjabri committed:
Merge pull request #1503 from dejan-brkic/bugfix/bpdescriptor

Blueprint Descriptor updates

Commit:
Added audit log for user and group management actions

jvega190 committed:
Merge branch 'develop' of https://github.com/craftercms/studio into feature/2839

jvega190 committed:
#2839 - Update from PR review

sumerjabri committed:
Merge pull request #1506 from dejan-brkic/bugfix/2933

CRAFTERCMS-2933

sumerjabri committed:
Merge pull request #1504 from jvega190/feature/2839

[studio-ui] Upgrade to TinyMCE 5 #2839

Commit:
merge with develop

Commit:
merge

dejan-brkic commented:
fixed PR

sumerjabri commented:
Closing because it's a bit messy with too many merges.
